File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/Extractors/INumberWithUnitExtractorConfiguration.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Collections.Generic;
using System.Collections.Immutable;
using System.Globalization;
using System.Linq;
using System.Text.RegularExpressions;

namespace Microsoft.Recognizers.Text.NumberWithUnit
{
    public interface INumberWithUnitExtractorConfiguration
    {
        ImmutableDictionary<string, string> SuffixList { get; }

        ImmutableDictionary<string, string> PrefixList { get; }

        ImmutableList<string> AmbiguousUnitList { get; }

        string ExtractType { get; }

        CultureInfo CultureInfo { get; }

        IExtractor UnitNumExtractor { get; }

        string BuildPrefix { get; }

        string BuildSuffix { get; }

        string ConnectorToken { get; }

        Regex CompoundUnitConnectorRegex { get; }

        Regex NonUnitRegex { get; }

        Regex MultiplierRegex { get; }

        Regex AmbiguousUnitNumberMultiplierRegex { get; }

        Dictionary<Regex, Regex> AmbiguityFiltersDict { get; }

        Dictionary<Regex, Regex> TemperatureAmbiguityFiltersDict { get; }

        Dictionary<Regex, Regex> DimensionAmbiguityFiltersDict { get; }

        void ExpandHalfSuffix(string source, ref List<ExtractResult> result, IOrderedEnumerable<ExtractResult> numbers);
    }
}

File: Java/libraries/recognizers-text-number-with-unit/src/main/java/com/microsoft/recognizers/text/numberwithunit/extractors/INumberWithUnitExtractorConfiguration.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.microsoft.recognizers.text.numberwithunit.extractors;

import com.microsoft.recognizers.text.CultureInfo;
import com.microsoft.recognizers.text.ExtractResult;
import com.microsoft.recognizers.text.IExtractor;

import java.util.List;
import java.util.Map;
import java.util.regex.Pattern;

public interface INumberWithUnitExtractorConfiguration {

    Map<String, String> getSuffixList();
    
    Map<String, String> getPrefixList();
    
    List<String> getAmbiguousUnitList();
    
    String getExtractType();
    
    CultureInfo getCultureInfo();
    
    IExtractor getUnitNumExtractor();
    
    String getBuildPrefix();
    
    String getBuildSuffix();
    
    String getConnectorToken();
    
    Pattern getCompoundUnitConnectorRegex();
    
    Pattern getAmbiguousUnitNumberMultiplierRegex();

    Map<Pattern, Pattern> getAmbiguityFiltersDict();

    List<ExtractResult> expandHalfSuffix(String source, List<ExtractResult> result, List<ExtractResult> numbers);
}
